File: src/main/java/com/meldmy/FindOdd.java
package com.meldmy;

import java.util.stream.IntStream;

import static java.util.function.Function.identity;
import static java.util.stream.Collectors.groupingBy;

/**
 * @author Dmytro Melnychuk
 */
public class FindOdd {
    public static int findIt(int[] a) {
        return IntStream.of(a)
                .boxed()

                .collect(groupingBy(identity()))
                .entrySet()
                .stream()
                .filter(i -> i.getValue().size() % 2 != 0)
                .findFirst().get().getKey();
    }
}

File: src/main/scala/com/meldmy/FindOdd.scala
package com.meldmy

/**
  * @author Dmytro Melnychuk
  */
class FindOdd {
  def findIt(a: Array[Int]): Int = a
    .groupBy(i => i)
    .filter(i => i._2.length % 2 != 0)
    .keys
    .last
}
